proposals/lip-Introduce_PoA_module.md

```diff
@@ -493,20 +492,17 @@ This function returns an array of bytes with the re-ordered list of addresses.
 ##### Execution
 
 ```python
-shuffleValidatorsList(validatorsAddresses, randomSeed)
+shuffleValidatorsList(validatorsAddresses, randomSeed):
     
-    validatorsList = []
-    for every address in validatorsAddresses:
-        validator.address = address
-        validator.roundHash = hash(randomSeed || address)
-        push validator to validatorsList
+    roundHash = {}
+    for address in validatorsAddresses:
+        roundHash[address] = hash(randomSeed || address)
     
     # Reorder the validator list in lexicographical order first by roundHash, then by address
-    sort validatorsList in increasing order where validator1 < validator2 if 
-        (validator1.roundHash < validator2.roundHash) or 
-        ((validator1.roundHash == validator2.roundHash) and (validator1.address < validator2.address))
+    shuffledValidatorAddresses = address in validatorsAddresses reordered by roundHash[address] 
+                                 ties broken lexicographically by address         
```

Comment:
This code is a bit odd and not very precise.
- validatorAddresses is already an array of addresses, so I don't understand why we write `address in validatorsAddresses`
- it does not say whether the sorting is increasing/decreasing
Suggestion:
```
  sort validatorsAddresses lexicogarphically by roundHash[address] breaking ties lexicographically by address  
```
or more specific
```
sort validatorsAddresses where address1 < adress2 if  (roundHash(address1) < roundHash(address2)) or 
    (( roundHash(address1) == roundHash(address2)) and address1 < address2)
```

Reply:
Changed as suggested. 

I prefer to return a different variable just to avoid side effects with input modification.

Reply:
Looks good. Can be resolved.